errors.go

```diff
@@ -283,6 +284,20 @@ func (errorType ErrorType) Set(err error) error {
 	}
 }
 
+// Set the type of the error
+func (errorType ErrorType) SetUserMessage(err error, msg string) error {
```

Comment:
Do we want to add a guard against empty messages?
```go
if msg == "" {
    return err
}
```

Reply:
Good point.
Since this is an external lib, I can't predict what the user will want to set, so I want to keep this option available - it basically means overriding the existing message with empty message.

Reply:
Fair enough :+1:

Reply:
Fair enough :+1:

Reply:
I tend to agree with the current approach.
We could add a `ClearUserMessage(error)` to do that exact thing, but I think we better keep it the way it is.